activiti-core/activiti-spring-boot-starter/src/main/java/org/activiti/spring/boot/CandidateStartersDeploymentConfigurer.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2010-2020 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.activiti.spring.boot;
+
+import org.activiti.bpmn.model.Process;
+import org.activiti.engine.impl.bpmn.deployer.BpmnDeploymentHelper;
+import org.activiti.engine.impl.context.Context;
+import org.activiti.engine.impl.persistence.entity.ProcessDefinitionEntity;
+import org.activiti.spring.SpringProcessEngineConfiguration;
+
+import java.util.List;
+
+public class CandidateStartersDeploymentConfigurer implements ProcessEngineConfigurationConfigurer {
+
+    private static final String EVERYONE_GROUP = "*";
+
+    @Override
+    public void configure(SpringProcessEngineConfiguration processEngineConfiguration) {
+        processEngineConfiguration.setBpmnDeploymentHelper(new CandidateStartersDeploymentHelper());
+    }
+
+    public class CandidateStartersDeploymentHelper extends BpmnDeploymentHelper {
+        @Override
+        public void addAuthorizationsForNewProcessDefinition(Process process, ProcessDefinitionEntity processDefinition) {
+            super.addAuthorizationsForNewProcessDefinition(process, processDefinition);
+            if (process != null &&
+                !process.isCandidateStarterUsersExists() &&
+                !process.isCandidateStarterGroupsExists()) {
+                addAuthorizationsFromIterator(Context.getCommandContext(), List.of(EVERYONE_GROUP), processDefinition, ExpressionType.GROUP);
+            }
+        }
```

Comment:
We need to double check what happens when an application is upgraded or have the runtime updated: I think the processes will get redeployed with a new version, but please, make sure that this is covered during the verification of this ticket.

Reply:
Yes thats what I think too. The process will be redeployed and identity links will be inserted. That should sync the query service too. But you are right we need to double check that.